src/rules/maxLineLengthRule.ts

```diff
@@ -49,15 +68,28 @@ export class Rule extends Lint.Rules.AbstractRule {
     }
 
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-        return this.applyWithFunction(sourceFile, walk, this.ruleArguments[0] as number);
+        return this.applyWithFunction(sourceFile, walk, {
+            ignoreUrls: this.ruleArguments.indexOf(OPTION_IGNORE_URL) > -1,
+            limit: this.ruleArguments[0] as number,
+        });
     }
 }
 
-function walk(ctx: Lint.WalkContext<number>) {
-    const limit = ctx.options;
+function walk(ctx: Lint.WalkContext<Options>) {
     for (const line of getLineRanges(ctx.sourceFile)) {
-        if (line.contentLength > limit) {
-            ctx.addFailureAt(line.pos, line.contentLength, Rule.FAILURE_STRING_FACTORY(limit));
+        if (line.contentLength > ctx.options.limit) {
+            if (ctx.options.ignoreUrls) {               //
+                if (hasUrl(line, ctx.sourceFile)) {     // To minimize regex searches
```

Comment:
can be combined with the enclosing condition

Reply:
Won't this search for URLs even if user passes `ignore-urls` option?